### ISSUE TITLE
Add option to disable certain mods via JSON

### ISSUE THREAD

**Issue by ithinkandicode** [GodotModding/godot-mod-loader#109]
We could add support for checking a file called `mods.json` in the user data folder, and when ModLoader inits, we could check that file and skip any disabled mods.

We could also provide a func to save the settings to this file.

This would let games implement a toggle to enable/disable certain mods from within the game itself, so users don't need to unsubscribe from them in the workshop/Thunderstore, or manually delete/rename mod ZIPs from their game folder.

**Comment by Qubus0** [comment]
agree. we can salvage a bit from #33 - [mod_loader](https://github.com/GodotModding/godot-mod-loader/pull/33/files#diff-d98477fdf59af4e0878f2040fe6458fd99a592b924b75d703f1e610c528cf063R448) and [mod_loader_helper](https://github.com/GodotModding/godot-mod-loader/pull/33/files#diff-2c11aaf064bd9e20f4165b366caf18abbd76d4bd136c12fff3e51b2b5da9ae38R89-R106), but just saving to a json instead.

**Comment by otDan** [comment]
I mean that is why I use thunderstore, you have separate profiles, don't think steam has that

**Comment by ithinkandicode** [comment]
> `mods-[profileName].json`

Best to keep it in one file. Then you can handle load/save with a single read/write operation. Eg:

```
{
	currentProfile: "Default",
	profiles: {
		"Default": {
			"Author-ModName": true,
			"Darkly77-ContentLoader": false,
		}
	}
}
```

Funcs we'll need to provide:
```
func load_mod_options() # private, loads options into ModLoader sinngleton's data on ModLoader init
func save_mod_options() # public, saves all data that's currently in memory, needs to be run after the following 2 funcs too

func enable_mod( modName: String, profileName: String = ""  ) # profileName is optional, defaults to current profile
func disable_mod( modName: String, profileName: String = ""  ) # same as above

func create_mod_options_profile( profileName: String ) # public
func change_mod_options_profile( profileName: String ) # public
```

**Comment by Qubus0** [comment]
Yes those are good. Should we enable any dependency of a mod when it is enabled? And disable mods if they depend on a mod that is being disabled?

**Comment by ithinkandicode** [comment]
I'm assigning myself to this as I'll be working with JSON saving for the Config JSON issues

**Comment by KANAjetzt** [comment]
I think its a good idea to add an extra key `mod_list` ( or something like that ), so we can extend the profiles in the future.

```json
{
	"current_profile": "Default",
	"profiles": {
		"Default": {
			"mod_list": {
				"Author-ModName": true,
				"Darkly77-ContentLoader": false
			}
		}
	}
}
```

**Comment by ithinkandicode** [comment]
- Closed by: #214
